File: 1392-find-the-difference-of-two-arrays/1392-find-the-difference-of-two-arrays.java
class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        Set<Integer> set = new HashSet<>();
        Set<Integer> set2 = new HashSet<>();

        for (int i=0; i<nums1.length; ++i) {
            set.add(nums1[i]);
        }

        for (int i=0; i<nums2.length; ++i) {
            set2.add(nums2[i]);
        }

        List<Integer> innerList1 = new ArrayList<>();
        List<Integer> innerList2 = new ArrayList<>();

        for (int num : set) {
            if (!set2.contains(num)) {
                innerList1.add(num);
            }
        }

        for (int num : set2) {
            if (!set.contains(num)) {
                innerList2.add(num);
            }
        }
        

        return Arrays.asList(innerList1, innerList2);
    }
}

File: 1392-find-the-difference-of-two-arrays/1392-find-the-difference-of-two-arrays.py
class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        set1 = set(nums1)
        set2 = set(nums2)

        list1 = []
        list2 = []
        
        for num in set1:
            if num not in set2:
                list1.append(num)
        
        for num in set2:
            if num not in set1:
                list2.append(num)

        return [list1, list2]